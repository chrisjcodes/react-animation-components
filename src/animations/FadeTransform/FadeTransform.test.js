import React from 'react';
import FadeTranform from './index.jsx';
import renderer from 'react-test-renderer';

describe('FadeTranform', () => {
    test('renders', () => {
        const component = renderer.create(<FadeTranform />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('can accept custom styles', () => {
        const component = renderer.create(
            <FadeTranform style={{ background: 'black' }} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionDelay with delay prop', () => {
        const component = renderer.create(<FadeTranform delay={1000} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionDuration with duration prop', () => {
        const component = renderer.create(<FadeTranform duration={1000} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionTimingFunction with timingFn prop', () => {
        const component = renderer.create(
            <FadeTranform timingFn="ease-in-out" />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets tranformProps', () => {
        const component = renderer.create(
            <FadeTranform
                transformProps={{ enterTransform: 'translateX(10em)' }}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets fadeProps', () => {
        const component = renderer.create(
            <FadeTranform fadeProps={{ enterOpacity: 0.85 }} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
