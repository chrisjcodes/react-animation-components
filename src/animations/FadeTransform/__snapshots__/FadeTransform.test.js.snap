// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeTranform can accept custom styles 1`] = `
<div
  className={undefined}
  style={
    Object {
      "background": "black",
      "opacity": 0,
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease",
      "willChange": "opacity",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "background": "black",
        "transform": "none",
        "transitionDelay": "0ms",
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "ease",
        "willChange": "transform",
      }
    }
  />
</div>
`;

exports[`FadeTranform renders 1`] = `
<div
  className={undefined}
  style={
    Object {
      "opacity": 0,
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease",
      "willChange": "opacity",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "transform": "none",
        "transitionDelay": "0ms",
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "ease",
        "willChange": "transform",
      }
    }
  />
</div>
`;

exports[`FadeTranform sets fadeProps 1`] = `
<div
  className={undefined}
  style={
    Object {
      "opacity": 0,
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease",
      "willChange": "opacity",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "transform": "none",
        "transitionDelay": "0ms",
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "ease",
        "willChange": "transform",
      }
    }
  />
</div>
`;

exports[`FadeTranform sets tranformProps 1`] = `
<div
  className={undefined}
  style={
    Object {
      "opacity": 0,
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease",
      "willChange": "opacity",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "transform": "none",
        "transitionDelay": "0ms",
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "ease",
        "willChange": "transform",
      }
    }
  />
</div>
`;

exports[`FadeTranform sets transitionDelay with delay prop 1`] = `
<div
  className={undefined}
  style={
    Object {
      "opacity": 0,
      "transitionDelay": "1000ms",
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease",
      "willChange": "opacity",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "transform": "none",
        "transitionDelay": "1000ms",
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "ease",
        "willChange": "transform",
      }
    }
  />
</div>
`;

exports[`FadeTranform sets transitionDuration with duration prop 1`] = `
<div
  className={undefined}
  style={
    Object {
      "opacity": 0,
      "transitionDelay": "0ms",
      "transitionDuration": "1000ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease",
      "willChange": "opacity",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "transform": "none",
        "transitionDelay": "0ms",
        "transitionDuration": "1000ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "ease",
        "willChange": "transform",
      }
    }
  />
</div>
`;

exports[`FadeTranform sets transitionTimingFunction with timingFn prop 1`] = `
<div
  className={undefined}
  style={
    Object {
      "opacity": 0,
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease-in-out",
      "willChange": "opacity",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "transform": "none",
        "transitionDelay": "0ms",
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "ease-in-out",
        "willChange": "transform",
      }
    }
  />
</div>
`;
